File: Expense-Income-tracker/src/main/java/com/pdevCode/Expense_Income_tracker/services/income/IncomeService.java
package com.pdevCode.Expense_Income_tracker.services.income;

import com.pdevCode.Expense_Income_tracker.dto.IncomeDto;
import com.pdevCode.Expense_Income_tracker.entities.Income;

import java.util.List;

public interface IncomeService {

    public Income postIncome(IncomeDto incomeDto);

    public List<IncomeDto> getAllIncomes();

    public Income updateIncome(Long id, IncomeDto incomeDto);

    public IncomeDto getById(Long id);

    public void deleteIncome(Long id);




    }

File: IncomeAndExpenseFrontend/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

const BASIC_URL = environment.url;

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(private http: HttpClient) { }

  postIncome(incomeDto: any): Observable<any>{
    return this.http.post(BASIC_URL + "api/income",  incomeDto);
  }

  getAllIncomes(): Observable<any>{
    return this.http.get(BASIC_URL  +"api/income/all");
  }

  deleteIncome(id:number): Observable<any>{
    return this.http.delete(BASIC_URL+ `api/income/${id}` );
  }


  getAllIncomeById(id:number):Observable<any>{
    return this.http.get(BASIC_URL+ `api/income/${id}`);
   }

   updateIncome(id:number, incomeDto: any):Observable<any>{
    return this.http.put(BASIC_URL+ `api/income/${id}`, incomeDto);
   }
}
